src/fides/api/util/tcf/tcf_experience_contents.py

```diff
@@ -149,12 +143,80 @@ class TCFExperienceContents(
     tcf_system_relationships: List[TCFVendorRelationships] = []
 
 
-def get_matching_privacy_declarations(db: Session) -> Query:
-    """Returns flattened system/privacy declaration records where we have a matching gvl data use AND the
-    legal basis for processing is "Consent" or "Legitimate interests"
+def get_legal_basis_override_subquery(db: Session) -> Alias:
+    """Subquery that allows us to globally override a purpose's legal basis for processing.
+
+    Original legal basis for processing is returned where:
+    - feature is disabled
+    - declaration's legal basis is not flexible
+    - no legal basis override specified
+
+    Null is returned where:
+    - Purpose is excluded (this will effectively remove the purpose from the Experience)
+
+    Otherwise, we return the override!
+    """
+    if not CONFIG.consent.override_vendor_purposes:
+        return db.query(
+            PrivacyDeclaration.id,
+            PrivacyDeclaration.legal_basis_for_processing.label(
+                "overridden_legal_basis_for_processing"
+            ),
+        ).subquery()
 
-    Only systems that meet this criteria should show up in the TCF overlay.
+    return (
+        db.query(
+            PrivacyDeclaration.id,
+            case(
+                [
+                    (
+                        PrivacyDeclaration.flexible_legal_basis_for_processing.is_(
+                            False
+                        ),
+                        PrivacyDeclaration.legal_basis_for_processing,
+                    ),
+                    (
+                        TCFPurposeOverride.is_included.is_(False),
+                        None,
+                    ),
+                    (
+                        TCFPurposeOverride.required_legal_basis.is_(None),
+                        PrivacyDeclaration.legal_basis_for_processing,
+                    ),
+                ],
+                else_=TCFPurposeOverride.required_legal_basis,
+            ).label("overridden_legal_basis_for_processing"),
+        )
+        .outerjoin(
+            TCFPurposeOverride,
+            TCFPurposeOverride.purpose == PrivacyDeclaration.purpose,
+        )
+        .subquery()
+    )
+
+
+def get_tcf_base_query_and_filters(
+    db: Session,
+) -> Tuple[Query, BinaryExpression, BinaryExpression]:
     """
+    Returns the base query that contains the foundations of the TCF Experience as well as
+    two filters to further refine the query when building the Experience.
```

Comment:
yeah, the returning these filters seems like probably the hardest thing here to easily digest, especially because it's going "across repos" 😬 

but that's ok, and having the special note here in the docstring helps!

Reply:
This could be alleviated some by moving more of this code over to Plus! There are some operations still here that need this information. I have a ticket in the backlog!